src/eth1spec/rlp.py

```diff
@@ -14,270 +14,346 @@
 
 from __future__ import annotations
 
-from typing import List, Sequence, Tuple, Union, cast
+from typing import List, Sequence, Union, cast
 
 from .base_types import U256, Uint
 from .eth_types import Bytes
+from .utils.common import does_idx_exist
 
-verbose = False
-debug = False
+RLP = Union[Bytes, Uint, U256, Sequence["RLP"]]  # type: ignore
 
 
-RLP = Union[Bytes, Uint, U256, Sequence["RLP"]]  # type: ignore
+#
+# RLP Encode
+#
 
 
-def encode(x: RLP) -> Bytes:
+def encode(raw_data: RLP) -> Bytes:
     """
-    Encodes `x` into a sequence of bytes using RLP.
+    Encodes `raw_data` into a sequence of bytes using RLP.
 
     Parameters
     ----------
-    x : `RLP`
-        A `Bytes`, `Uint`, or sequence of `RLP` encodable objects.
+    raw_data : `RLP`
+        A `Bytes`, `Uint`, `Uint256` or sequence of `RLP` encodable
+        objects.
 
     Returns
     -------
     encoded : `eth1spec.eth_types.Bytes`
-        The RLP encoded bytes representing `x`.
+        The RLP encoded bytes representing `raw_data`.
     """
-    if verbose:
-        print("RLP(", x, ")", "type: ", type(x))
-    if isinstance(x, (bytearray, bytes)):
-        return R_b(x)
-    elif isinstance(x, (Uint, U256)):
-        return encode(x.to_be_bytes())
-    elif isinstance(x, Sequence):
-        return R_l(cast(Sequence[RLP], x))
+    if isinstance(raw_data, (bytearray, bytes)):
```

Comment:
We could use Union, but I feel it would be unnecessary here. In such cases Union is generally not used. Is there any reason why you want Union to be here?

Reply:
Sounds good to me then!